test/framework/comparators.py

```diff
@@ -5,15 +5,15 @@ class AnyStringMatching(object):
     """
     A helper object that compares equal to any string matching the specified pattern.
     """
-    def __init__(self, pattern):
-        self._pattern = pattern
+    def __init__(self, regex_pattern):
+        self._matcher = re.compile(regex_pattern)
```

Comment:
I refactored this class so that the regex is checked in the constructor instead of during comparison. The exception that gets raised when an invalid regex is used is not clear at all, so checking the regex pattern in the constructor will at least give a more helpful stack trace.